optical_rl_gym/envs/rmcsa_env.py

```diff
@@ -320,14 +337,20 @@ def get_number_slots(self, path: Path) -> int:
         """
         return math.ceil(self.service.bit_rate / path.best_modulation['capacity']) + 1
 
-    def is_path_free(self, path: Path, initial_slot: int, number_slots: int) -> bool:
+    def is_path_free(self, core: int, path: Path, initial_slot: int, number_slots: int) -> bool:
+        """
+        NEW DOC STRING, NOT FINAL
+
+        Method that determines if the path is free for the core, path, and initial_slot.
+        """
         if initial_slot + number_slots > self.num_spectrum_resources:
             # logging.debug('error index' + env.parameters.rsa_algorithm)
             return False
         for i in range(len(path.node_list) - 1):
             if np.any(self.topology.graph['available_slots'][
-                      self.topology[path.node_list[i]][path.node_list[i + 1]]['index'],
-                      initial_slot:initial_slot + number_slots] == 0):
+                      initial_slot:initial_slot + number_slots,
+                      core,
+                      self.topology[path.node_list[i]][path.node_list[i + 1]]['index']] == 0):
```

Comment:
If this path of implementation is correct, then we will be able to continue a significant amount of changes. Before we wanted to proceed though, we weren't sure if this was a proper implementation due to our questions.

Reply:
Our main issue came from this block of code. Trying to understand what this code does has been difficult, we understand its checking if any of the items inside equal to 0. What we're wondering about it though is for example, does each item (initial_slot, core, self.topology) perform an index operation on `self.topology.graph['available_slots']` and it checks if those evaluate to 0. Or does it check just if `core` it self evaluates to 0.